src/main/java/com/zachary_moore/runner/LintRunner.java

```diff
@@ -5,78 +5,74 @@
-import java.util.*;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.logging.Logger;
 import java.util.stream.Collectors;
+import org.apache.commons.io.FileUtils;
+
 
 public class LintRunner {
 
-    private List<JSONFile> filesToLint;
+    private String[] basePaths;
     private Map<LintRule, Map<JSONFile, List<String>>> lintOutput;
     private final LintRegister lintRegister;
 
     /**
      * Create a LintRunner
      * @param lintRegister representation of {@link LintRule} to run against
-     * @param basePath Base Path to setup {@link JSONFile} from
+     * @param basePaths Paths to folders or files to fetch {@link JSONFile}(s) from
      */
     public LintRunner(LintRegister lintRegister,
-                      String basePath) {
+                      String... basePaths) {
         this.lintRegister = lintRegister;
-
-        File basePathFile = new File(basePath);
-        if (basePathFile.isFile()) {
-            try {
-                this.filesToLint = new ArrayList<>(Collections.
-                        singletonList(new JSONFile(basePathFile)));
-            } catch (IOException e) {
-                throw new RuntimeException(e);
-            }
-        } else if (basePathFile.isDirectory()){
-            this.filesToLint = getAllFiles(basePath).stream()
-                    .map(f -> {
-                        try {
-                            return new JSONFile(f);
-                        } catch (IOException e) {
-                            throw new RuntimeException(e);
-                        }
-                    })
-                    .filter(Objects::nonNull)
-                    .collect(Collectors.toList());
-        } else {
-            throw new RuntimeException("Input path is neither a file nor directory");
-        }
+        this.basePaths = basePaths;
     }
 
-    private List<File> getAllFiles(String basePath) {
-        List<File> files = new ArrayList<>();
-        File directory = new File(basePath);
+    private Set<JSONFile> getAllFiles(String... basePaths) {
+        Set<File> files = new HashSet<>();
 
-        File[] fList = directory.listFiles();
-        if (fList != null) {
-            for (File file : fList) {
-                if (file.isFile()) {
-                    files.add(file);
-                } else if (file.isDirectory()) {
-                    files.addAll(getAllFiles(file.getAbsolutePath()));
-                }
+        for (String basePath : basePaths) {
+            File file = new File(basePath);
+
+            if (file.isFile()) {
+                files.add(file);
+            } else if (file.isDirectory()) {
+                files.addAll(FileUtils.listFiles(file, null, true));
+            } else {
+                // TODO: Add proper logging
+                Logger.getGlobal().warning(basePath + " is not a valid file path");
```

Comment:
Would love feedback here. We don't want to fail the runner because one of the provided paths was wrong. But it should be more conspicuous than a log. Is it possible to include this failure as a part of the report?